records_mover/records/job/config.py

```diff
@@ -29,8 +29,13 @@ def non_default_arg_names(self, fn: Callable[..., Any]) -> List[str]:
 
     def add_hint_parameter(self, kwargs: Dict[str, Any], param_name: str) -> None:
         if 'records_format' in kwargs:
+            existing_records_format = kwargs['records_format']
+            if not isinstance(existing_records_format, DelimitedRecordsFormat):
+                raise NotImplementedError('Hints are not compatible '
+                                          'with records format '
+                                          f'{existing_records_format.format_type}')
```

Comment:
Since we now might get contradictory combination of arguments (like `--source.format parquet --source.delimiter '|'`, we need to check for that situation.

Note that one day ParquetRecordsFormat will support some very small number of hints (compression is a key one that it's missing today), but as of today, there are no hints it supports.